c10/core/GradMode.h

```diff
@@ -1,10 +1,16 @@
 #pragma once
 
+#include <c10/core/impl/ThreadLocalState.h>
 #include <c10/macros/Macros.h>
 
 namespace c10 {
 
 struct TORCH_API GradMode {
+  // NB: We inline this overload because it allows us to fully inline hot paths.
+  static inline bool is_enabled(const impl::PODLocalState* const tls) {
```

Comment:
Would you mind explaining why we want to pass tls as argument instead of just querying `_get_thread_local_state` inside the function body?

Reply:
Done.